### ISSUE TITLE
Unable to connect AML compute instance from Visual Studio Code (new error)

### ISSUE THREAD

**Issue by gf-eaton** [microsoft/vscode-tools-for-ai#2037]
Does this occur consistently? Yes, since this week. 5 days ago.
Repro steps:

1. right click connect or connect using new window
2. error will be seen `Failed to connect to the remote extension host server (Error: request to https://gf-a8mv2.eastus.instances.azureml.ms/api/terminals?1683743256805 failed, reason: self signed certificate in certificate chain)`

Action: Resolver.resolve
Error type: 70
Error Message: request to redacted:url failed, reason: self signed certificate in certificate chain

Version: 0.28.1
OS: win32
OS Release: 10.0.19044
Product: Visual Studio Code
Product Version: 1.78.1
Language: en

<details>
<summary>Call Stack</summary>

```
s extension.js:2:1981706
extension.js:2:2007388extension.js:2:2007388
```

</details>

**Comment by gf-eaton** [comment]
#2033 Same error

**Comment by gf-eaton** [comment]
In Visual Studio Code : Right click start and stop compute instance works... 
In the other hand Connect or Connect In new Window are producing the error.

I use Visual Studio Code everyday, this is new error, was not doing this last week ...

**Comment by alexweininger** [comment]
A change released in the VS Code 1.78 April release breaks loading OS certificates on Windows. Meaning self-signed certificates and CA certificates picked up from a network proxy are currently broken. See https://github.com/microsoft/vscode/issues/181404 for more details.

It looks like VS Code is planning to include a fix in the [1.78.2 release (April recovery 2)](https://github.com/microsoft/vscode/issues/181866).

Until that is released, you can downgrade to VS Code version 1.77. Also, [other users](https://github.com/microsoft/vscode-azure-account/issues/792#issuecomment-1540127373) have had success installing the [win-ca extension](https://marketplace.visualstudio.com/items?itemName=ukoloff.win-ca), which makes Trusted Root Certificates on Windows available to VS Code extensions.

Another workaround if your CA certificate is provided by a network proxy is to configure the proxy related settings in VS Code.

**Comment by sevillal** [comment]
@alexweininger thank for pointing that out. 

@gf-eaton could you please try downgrading or installing the win-ca extension and let me know it that help.

**Comment by gf-eaton** [comment]
Hello all,

@sevillal Thanks for the replies.
@alexweininger Thanks it confirms our founding's.

### Unfortunate 
`win-ca` **is not** resolving the issue.
Here is the popup message in the lower corner : `request to https://gf-ds12v2.eastus.instances.azureml.ms/api/terminals?1683806412459 failed, reason: self signed certificate in certificate chain`

### Not an option
As per downgrading Visual Studio Code we cannot do that now.

### Solution 💯 
We are now using the proxy workaround : in `Visual Studio Code > Settings > Proxy` : https://myproxy.corpo.net:8080 and also SSL check is `OFF`

**Comment by sevillal** [comment]
@gf-eaton VS Code just released version 1.78.2 which should fix the certificates issue, could you please update VS Code (Help > Check for Updates...) and retry the connection.

**Comment by gf-eaton** [comment]
@sevillal 1.78.2 is now functional, fix the issue.
